pkg/plugins/golang/klogr/v1/init.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+)
+
+var _ plugin.InitSubcommand = &initSubcommand{}
+
+type initSubcommand struct {
+	config config.Config
```

Comment:
You are implementing here the INIT interface which means that the changes applied here would only be called when the project is init ( kb init command )

In this case, the logs are scaffold when we create the API then, which means that this implementation would not work. More info: https://book.kubebuilder.io/plugins/plugins.html

Also, the plugins added to the project starts to be supported by it and we need to discuss if we would like or not to have this plugin by default. 

It is an important to highlight as well that the first version of the plugin should be alpha since it would not be mature enough.  See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/VERSIONING.md

Reply:
@justinsb is that the expected behaviour? would we only want this change when we run the init command with this plugin or should it work with creating API as well?